### DISCUSSION TITLE
Theme reverting back to initial theme

### DISCUSSION THREAD

**Discussion by shahzaib78631** [jpudysz/react-native-unistyles#441] [Q&A] [answered]
Hello , 

I would like to know if i am doing anything wrongly when changing the theme. I am using `KeyboardAwareScrollView` from `react-native-keyboard-controller` following is the minimal code.

```
const ThemedKeyboardAwareScrollView = withUnistyles(KeyboardAwareScrollView)
<ThemedKeyboardAwareScrollView style={styles.container}>{children}</ThemedKeyboardAwareScrollView>
```
```
const styles = StyleSheet.create((theme) => ({
  container: { 
    backgroundColor: theme.colors.background
  })
);
```
When i change the theme using `UnistylesRuntime.setTheme(name)` It changes the theme to the provided themeName for the active screen. But when i navigate to another screen then the theme reverts back to the initialTheme defined in the Stylesheet.configure.

Before changing theme

![simulator_screenshot_C51D805F-0A24-4C07-AEEA-983A6FF98909](https://github.com/user-attachments/assets/b017ff62-2730-4bb2-aa25-f472954e16b3)

After changing theme

![simulator_screenshot_2E3D639D-BFA5-42A9-9EE3-CE1590AA5BE7](https://github.com/user-attachments/assets/f75c9c5f-30a5-4d80-9a7e-898b07a379a5)

After navigating and return

![simulator_screenshot_1778D707-AA8A-4E14-AE8D-9128AA526C00](https://github.com/user-attachments/assets/93ccd5e3-1fa1-4023-8094-701528a08bcd)

Same is true for some other components.

**Comment by jpudysz** [comment]
Your code looks correct, so it's most likely a bug. Do you have a repo to share, or is this a commercial project? 
It's just easier to debug when I have the whole example. If not, that's fine too, I'll try to reproduce it locally

**Reply by shahzaib78631** [reply]
Yes you can check my following [repo](https://github.com/shahzaib78631/spliit-mobile-app).

File Path: components/ui/ThemedView.tsx

There are some more points that i have noticed:

1: Issue with popup menu. Using 
`withUnistyles(MenuOptions)` does not work. Getting error that `Menu should have at least one MenuOptions.` 

File path: components/base/BasePopupMenu.tsx

2: Using `useAnimatedStyles` or applying Styles on `Animated.View` doesn't work.

File path: components/CustomBottomTabs.tsx

3: Applying inline styles to a `View` which have `StyleSheet` styles applied from `unistyles` doesn't update theme or other applied styles. I think this relates to [Issue-368](https://github.com/jpudysz/react-native-unistyles/issues/368).

**Question**: How to change placeholder color according to the active theme color. This also applies to Components which use color property i.e MaterialIcons etc. 

```
<TextInput
          style={[styles.input, commonStyles.flex1, inputStyle]}
          placeholder={placeholder}
          placeholderTextColor={color} <--- Change This Color
          value={value}
          onChangeText={onChangeText}
          onFocus={() => setIsFocused(true)}
          onBlur={() => setIsFocused(false)}
          {...rest}
        />
 ```
 If we use `const UniTextInput =  withUnistyles(TextInput)`  then it throws error for 
 
 `style={[styles.input, commonStyles.flex1, inputStyle]}` 
 Unistyles: withUnistyles requires style to be an object. Please check props for component: undefined [Component Stack]
 
 `style={{ ...styles.input, ...commonStyles.flex1 }}`
  Unistyles: withUnistyles received style that is not bound. You likely used the spread operator on a Unistyle style. Please check props for component: undefined [Component Stack]
  
  **Note**: Using a single style object i.e `style={styles.input}` works completely fine when used with `withUnistyles`

**Reply by jpudysz** [reply]
Thank you @shahzaib78631 !
Let's test it against `beta.3`  before we create any new issues.

Beta 3 should land soon

**Reply by jpudysz** [reply]
Checking your repo now

**Reply by jpudysz** [reply]
Theme switching looks good:
https://github.com/user-attachments/assets/8092e8c8-0aac-4417-96fc-72a1ea09b0bc

**Reply by jpudysz** [reply, marked as answer]
> Question: How to change placeholder color according to the active theme color. This also applies to Components which use color property i.e MaterialIcons etc.

1: Issue with popup menu - wrapped it, no more issues, but styles are somehow broken, not sure how react-native-popup-menu works 
2. Yeah, I'm aware of it. I will dedicate beta.6 for Reanimated support, for now there is no workaround
3. If you're using React compiler, then it's most likely it 

>If we use const UniTextInput =  withUnistyles(TextInput) then it throws error for
>
>style={[styles.input, commonStyles.flex1, inputStyle]}
>Unistyles: withUnistyles requires style to be an object. Please check props for component: undefined [Component Stack]
>
>style={{ ...styles.input, ...commonStyles.flex1 }}
>Unistyles: withUnistyles received style that is not bound. You likely used the spread operator on a Unistyle style. Please check >props for component: undefined [Component Stack]


Yeah, withUnistyles is limited to single style now. In beta.5 I will remove `Unistyles is not bound!` so it should work as expected.

So most likely your issues will be resolved with beta.6.

**Reply by shahzaib78631** [reply]
> Theme switching looks good: https://github.com/user-attachments/assets/8092e8c8-0aac-4417-96fc-72a1ea09b0bc

Yes i also checked against beta 4 it seems to be working fine now.
